flashlight/fl/tensor/backend/jit/eval/Evaluator.cpp

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include "flashlight/fl/tensor/backend/jit/eval/Evaluator.h"
+
+#include "flashlight/fl/tensor/backend/jit/JitTensorBase.h"
+#include "flashlight/fl/tensor/backend/jit/ir/ValueNode.h"
+
+namespace fl {
+
+namespace {
+
+void fillNodeRefCountInTree(
+    Node* root,
+    std::unordered_map<Node*, unsigned>& nodeToRefCount) {
+  if (nodeToRefCount.find(root) == nodeToRefCount.end()) {
+    nodeToRefCount.emplace(root, root->getRefCount());
+    for (const auto& input : root->inputs()) {
+      fillNodeRefCountInTree(input, nodeToRefCount);
```

Comment:
I'd sort of prefer this were iterative in preparation for having huge graphs, but either way

Reply:
Good point. I had this thought yesterday but forgot to update it.